src/libs/antares/study/parts/thermal/cluster.cpp

```diff
@@ -235,6 +234,29 @@ void Data::ThermalCluster::copyFrom(const ThermalCluster& cluster)
         parentArea->forceReload();
 }
 
+static Data::ThermalCluster::ThermalDispatchableGroup stringToGroup(const Data::ClusterName& newgrp)
+{
+    using namespace Antares::Data;
+    const static std::map<ClusterName, ThermalCluster::ThermalDispatchableGroup> mapping
+      = {{"nuclear", ThermalCluster::thermalDispatchGrpNuclear},
+         {"lignite", ThermalCluster::thermalDispatchGrpLignite},
+         {"hard coal", ThermalCluster::thermalDispatchGrpHardCoal},
+         {"gas", ThermalCluster::thermalDispatchGrpGas},
+         {"oil", ThermalCluster::thermalDispatchGrpOil},
+         {"mixed fuel", ThermalCluster::thermalDispatchGrpMixedFuel},
+         {"other", ThermalCluster::thermalDispatchGrpOther1},
+         {"other 1", ThermalCluster::thermalDispatchGrpOther1},
+         {"other 2", ThermalCluster::thermalDispatchGrpOther2},
+         {"other 3", ThermalCluster::thermalDispatchGrpOther3},
+         {"other 4", ThermalCluster::thermalDispatchGrpOther4}};
+    if (mapping.count(newgrp) > 0)
```

Comment:
Of course not a bottleneck as a whole, but this will go over all the collection to count, and we will perform another search below: it would be fore efficient to just search once with a `find`.